--- src/linux/drivers/gpu/drm/i915/display/intel_dmc.c
+++ src/linux/drivers/gpu/drm/i915/display/intel_dmc.c
@@ -1545,9 +1545,19 @@
 void intel_dmc_update_dc6_allowed_count(struct intel_display *display,
 					bool start_tracking)
 {
+	if (!display) {
+		printk("%s:%u DISPLAY NULL POINTER\n", __func__, __LINE__);
+		return;
+	}
+
 	struct intel_dmc *dmc = display_to_dmc(display);
 	u32 dc5_cur_count;
 
+	if (!dmc) {
+		printk("%s:%u DMC NULL POINTER\n", __func__, __LINE__);
+		return;
+	}
+
 	if (DISPLAY_VER(dmc->display) < 14)
 		return;
 
@@ -1565,6 +1575,11 @@
 	struct intel_dmc *dmc = display_to_dmc(display);
 	bool dc6_enabled;
 
+	if (!dmc) {
+		printk("%s:%u DMC NULL POINTER\n", __func__, __LINE__);
+		return false;
+	}
+
 	if (DISPLAY_VER(display) < 14)
 		return false;
 
@@ -1643,8 +1658,11 @@
 		seq_printf(m, "DC5 -> DC6 count: %d\n",
 			   intel_de_read(display, dc6_reg));
 
+	if (dmc)
 	seq_printf(m, "program base: 0x%08x\n",
 		   intel_de_read(display, DMC_PROGRAM(dmc->dmc_info[DMC_FW_MAIN].start_mmioaddr, 0)));
+	else
+	seq_printf(m, "program base: unknown");
 
 out:
 	seq_printf(m, "ssp base: 0x%08x\n",
