--- src/linux/drivers/gpu/drm/xe/xe_bo.c
+++ src/linux/drivers/gpu/drm/xe/xe_bo.c
@@ -2647,7 +2675,15 @@
 struct xe_bo *xe_managed_bo_create_from_data(struct xe_device *xe, struct xe_tile *tile,
 					     const void *data, size_t size, u32 flags)
 {
-	struct xe_bo *bo = xe_managed_bo_create_pin_map(xe, tile, ALIGN(size, PAGE_SIZE), flags);
+	size_t asize = size;
+	size_t hbit = 63 - __builtin_clzl(asize | 1);
+
+	if ((1u << hbit) < size) {
+		asize = 1u << (hbit + 1);
+		printk("%s:%u - resize %lx->%lx for contiguous vmap\n", __func__, __LINE__, size, asize);
+	}
+
+	struct xe_bo *bo = xe_managed_bo_create_pin_map(xe, tile, ALIGN(asize, PAGE_SIZE), flags);
 
 	if (IS_ERR(bo))
 		return bo;
@@ -3592,7 +3592,15 @@
 	args->size = ALIGN(mul_u32_u32(args->pitch, args->height),
 			   page_size);
 
-	bo = xe_bo_create_user(xe, NULL, args->size,
+	size_t asize = args->size;
+	size_t hbit = 63 - __builtin_clzl(asize | 1);
+
+	if ((1u << hbit) < args->size) {
+		asize = 1u << (hbit + 1);
+		printk("%s:%u - resize %llx->%lx for contiguous vmap\n", __func__, __LINE__, args->size, asize);
+	}
+
+	bo = xe_bo_create_user(xe, NULL, asize,
 			       DRM_XE_GEM_CPU_CACHING_WC,
 			       XE_BO_FLAG_VRAM_IF_DGFX(xe_device_get_root_tile(xe)) |
 			       XE_BO_FLAG_SCANOUT |
--- src/linux/drivers/gpu/drm/xe/xe_guc_ads.c
+++ src/linux/drivers/gpu/drm/xe/xe_guc_ads.c
@@ -398,7 +398,15 @@
 	ads->regset_size = calculate_regset_size(gt);
 	ads->ads_waklv_size = calculate_waklv_size(ads);
 
-	bo = xe_managed_bo_create_pin_map(xe, tile, guc_ads_size(ads) + MAX_GOLDEN_LRC_SIZE,
+	size_t asize = guc_ads_size(ads) + MAX_GOLDEN_LRC_SIZE;
+	size_t hbit = 63 - __builtin_clzl(asize | 1);
+
+	if ((1u << hbit) < asize) {
+		asize = 1u << (hbit + 1);
+		printk("%s:%u - resize %lx->%lx for contiguous vmap\n", __func__, __LINE__, guc_ads_size(ads) + MAX_GOLDEN_LRC_SIZE, asize);
+	}
+
+	bo = xe_managed_bo_create_pin_map(xe, tile, asize,
 					  XE_BO_FLAG_SYSTEM |
 					  XE_BO_FLAG_GGTT |
 					  XE_BO_FLAG_GGTT_INVALIDATE |
--- src/linux/drivers/gpu/drm/xe/xe_guc_log.c
+++ src/linux/drivers/gpu/drm/xe/xe_guc_log.c
@@ -257,7 +257,15 @@
 	struct xe_tile *tile = gt_to_tile(log_to_gt(log));
 	struct xe_bo *bo;
 
-	bo = xe_managed_bo_create_pin_map(xe, tile, guc_log_size(),
+	size_t asize = guc_log_size();
+	size_t hbit = 63 - __builtin_clzl(asize | 1);
+
+	if ((1u << hbit) < asize) {
+		asize = 1u << (hbit + 1);
+		printk("%s:%u - resize %lx->%lx for contiguous vmap\n", __func__, __LINE__, guc_log_size(), asize);
+	}
+
+	bo = xe_managed_bo_create_pin_map(xe, tile, asize,
 					  XE_BO_FLAG_SYSTEM |
 					  XE_BO_FLAG_GGTT |
 					  XE_BO_FLAG_GGTT_INVALIDATE |
@@ -265,7 +273,7 @@
 	if (IS_ERR(bo))
 		return PTR_ERR(bo);
 
-	xe_map_memset(xe, &bo->vmap, 0, 0, guc_log_size());
+	xe_map_memset(xe, &bo->vmap, 0, 0, asize);
 	log->bo = bo;
 	log->level = xe_modparam.guc_log_level;
 
--- src/linux/drivers/gpu/drm/xe/xe_guc_ct.c
+++ src/linux/drivers/gpu/drm/xe/xe_guc_ct.c
@@ -276,7 +276,15 @@
 	struct xe_tile *tile = gt_to_tile(gt);
 	struct xe_bo *bo;
 
-	bo = xe_managed_bo_create_pin_map(xe, tile, guc_ct_size(),
+	size_t asize = guc_ct_size();
+	size_t hbit = 63 - __builtin_clzl(asize | 1);
+
+	if ((1u << hbit) < asize) {
+		asize = 1u << (hbit + 1);
+		printk("%s:%u - resize %lx->%lx for contiguous vmap\n", __func__, __LINE__, guc_ct_size(), asize);
+	}
+
+	bo = xe_managed_bo_create_pin_map(xe, tile, asize,
 					  XE_BO_FLAG_SYSTEM |
 					  XE_BO_FLAG_GGTT |
 					  XE_BO_FLAG_GGTT_INVALIDATE |
--- src/linux/drivers/gpu/drm/xe/xe_lrc.c
+++ src/linux/drivers/gpu/drm/xe/xe_lrc.c
@@ -1418,8 +1418,16 @@
 	if (vm && vm->xef) /* userspace */
 		bo_flags |= XE_BO_FLAG_PINNED_LATE_RESTORE;
 
+	u32 asize = bo_size;
+	size_t hbit = 31 - __builtin_clz(asize | 1);
+
+	if ((1u << hbit) < asize) {
+		asize = 1u << (hbit + 1);
+		printk("%s:%u - resize %x->%x for contiguous vmap\n", __func__, __LINE__, bo_size, asize);
+	}
+
 	lrc->bo = xe_bo_create_pin_map_novm(xe, tile,
-					    bo_size,
+					    asize,
 					    ttm_bo_type_kernel,
 					    bo_flags, false);
 	if (IS_ERR(lrc->bo))
